app/Allocator/Squawk/General/CcamsSquawkAllocator.php

```diff
@@ -53,9 +51,32 @@ public function fetch(string $callsign): ?SquawkAssignmentInterface
         return CcamsSquawkAssignment::find($callsign);
     }
 
-    public function canAllocateForCategory(string $category): bool
+    public function assignToCallsign(string $code, string $callsign): ?SquawkAssignmentInterface
     {
-        return $category === SquawkAssignmentCategories::GENERAL;
+        $assignment = null;
+        CcamsSquawkRange::all()->each(
```

Comment:
I can't see any specific changes to this common block (57-77) throughout these allocator files. Would it be best to put into the AbstractSquawkAllocator class and pass in the range and assignment classes to prevent duplicate logic?

Reply:
Had a look but it gets really complex, so I think a bit of duplication is ok here.